lib/commands/access.js

```diff
@@ -1,223 +1,225 @@
-    if (!perms || (perms !== 'read-only' && perms !== 'read-write')) {
-      throw this.usageError('First argument must be either `read-only` or `read-write`.')
-    }
-
-    if (!scopeteam) {
-      throw this.usageError('`<scope:team>` argument is required.')
-    }
-
-    const [, scope, team] = scopeteam.match(/^@?([^:]+):(.*)$/) || []
-
-    if (!scope && !team) {
-      throw this.usageError(
-        'Second argument used incorrect format.\n' +
-        'Example: @example:developers'
-      )
-    }
-
-    return this.modifyPackage(pkg, opts, (pkgName, opts) =>
-      libaccess.grant(pkgName, scopeteam, perms, opts), false)
-  }
-
-  async revoke ([scopeteam, pkg], opts) {
-    if (!scopeteam) {
-      throw this.usageError('`<scope:team>` argument is required.')
-    }
-
-    const [, scope, team] = scopeteam.match(/^@?([^:]+):(.*)$/) || []
-
-    if (!scope || !team) {
-      throw this.usageError(
-        'First argument used incorrect format.\n' +
-        'Example: @example:developers'
-      )
+    switch (cmd) {
+      case 'grant':
+        // npm access grant read-only scope:team [package]
+        if (!['read-only', 'read-write'].includes(subcmd)) {
+          throw this.usageError('grant must be either `read-only` or `read-write`')
+        }
+        if (!args[0]) {
+          throw this.usageError('`<scope:team>` argument is required')
+        }
+        return this.#grant(subcmd, args[0], args[1])
+      case 'revoke':
+        return this.#revoke(subcmd, args[0])
+      case 'list':
+      case 'ls':
+        if (subcmd === 'packages') {
+          return this.#listPackages(args[0], args[1])
+        }
+        if (subcmd === 'collaborators') {
+          return this.#listCollaborators(args[0], args[1])
+        }
+        throw this.usageError(`list ${subcmd} is not a valid access command`)
+      case 'get':
+        if (subcmd !== 'status') {
+          throw this.usageError(`get ${subcmd} is not a valid access command`)
+        }
+        return this.#getStatus(args[0])
+      case 'set':
+        if (!setCommands.includes(subcmd)) {
+          throw this.usageError(`set ${subcmd} is not a valid access command`)
+        }
+        return this.#set(subcmd, args[0])
     }
-
-    return this.modifyPackage(pkg, opts, (pkgName, opts) =>
-      libaccess.revoke(pkgName, scopeteam, opts))
-  }
-
-  get ['2fa-required'] () {
-    return this.tfaRequired
-  }
-
-  tfaRequired ([pkg], opts) {
-    return this.modifyPackage(pkg, opts, libaccess.tfaRequired, false)
   }
 
-  get ['2fa-not-required'] () {
-    return this.tfaNotRequired
+  async #grant (permissions, scope, pkg) {
+    await libnpmaccess.setPermissions(scope, pkg, permissions)
+    // TODO output?
   }
 
-  tfaNotRequired ([pkg], opts) {
-    return this.modifyPackage(pkg, opts, libaccess.tfaNotRequired, false)
+  async #revoke (scope, pkg) {
+    await libnpmaccess.removePermissions(scope, pkg)
+    // TODO output?
```

Comment:
Similar to the above, if the action was successful we shouldn't need to output any messaging (this is similar to subcommands which offer `delete` - ex. `npm config delete <key>`), just ensure the status code is set properly. If there's an error setting/revoking permissions then that would be when we should bubble up/throw an error to the end-user (ideally using messaging that aligns with whatever the returned http status code was sent back by the configured registry)